test/functional/p2p_private_broadcast.py

```diff
@@ -0,0 +1,425 @@
+from test_framework.socks5 import (
+    Socks5Configuration,
+    Socks5Server,
+)
+from test_framework.test_framework import (
+    BitcoinTestFramework,
+)
+from test_framework.util import (
+    MAX_NODES,
+    assert_equal,
+    assert_not_equal,
+    assert_raises_rpc_error,
+    p2p_port,
+    tor_port,
+)
+from test_framework.wallet import (
+    MiniWallet,
+)
+
+MAX_OUTBOUND_FULL_RELAY_CONNECTIONS = 8
+MAX_BLOCK_RELAY_ONLY_CONNECTIONS = 2
+NUM_INITIAL_CONNECTIONS = MAX_OUTBOUND_FULL_RELAY_CONNECTIONS + MAX_BLOCK_RELAY_ONLY_CONNECTIONS
+NUM_PRIVATE_BROADCAST_PER_TX = 3
+
+# Fill addrman with these addresses. Must have enough Tor addresses, so that even
+# if all 10 default connections are opened to a Tor address (!?) there must be more
+# for private broadcast.
+ADDRMAN_ADDRESSES = [
+    "1.65.195.98",
+    "2.59.236.56",
+    "2.83.114.20",
+    "2.248.194.16",
+    "5.2.154.6",
+    "5.101.140.30",
+    "5.128.87.126",
+    "5.144.21.49",
+    "5.172.132.104",
+    "5.188.62.18",
+    "5.200.2.180",
+    "8.129.184.255",
+    "8.209.105.138",
+    "12.34.98.148",
+    "14.199.102.151",
+    "18.27.79.17",
+    "18.27.124.231",
+    "18.216.249.151",
+    "23.88.155.58",
+    "23.93.101.158",
+
+    "[2001:19f0:1000:1db3:5400:4ff:fe56:5a8d]",
+    "[2001:19f0:5:24da:3eec:efff:feb9:f36e]",
+    "[2001:19f0:5:24da::]",
+    "[2001:19f0:5:4535:3eec:efff:feb9:87e4]",
+    "[2001:19f0:5:4535::]",
+    "[2001:1bc0:c1::2000]",
+    "[2001:1c04:4008:6300:8a5f:2678:114b:a660]",
+    "[2001:41d0:203:3739::]",
+    "[2001:41d0:203:8f49::]",
+    "[2001:41d0:203:bb0a::]",
+    "[2001:41d0:2:bf8f::]",
+    "[2001:41d0:303:de8b::]",
+    "[2001:41d0:403:3d61::]",
+    "[2001:41d0:405:9600::]",
+    "[2001:41d0:8:ed7f::1]",
+    "[2001:41d0:a:69a2::1]",
+    "[2001:41f0::62:6974:636f:696e]",
+    "[2001:470:1b62::]",
+    "[2001:470:1f05:43b:2831:8530:7179:5864]",
+    "[2001:470:1f09:b14::11]",
+
+    "2bqghnldu6mcug4pikzprwhtjjnsyederctvci6klcwzepnjd46ikjyd.onion",
+    "4lr3w2iyyl5u5l6tosizclykf5v3smqroqdn2i4h3kq6pfbbjb2xytad.onion",
+    "5g72ppm3krkorsfopcm2bi7wlv4ohhs4u4mlseymasn7g7zhdcyjpfid.onion",
+    "5sbmcl4m5api5tqafi4gcckrn3y52sz5mskxf3t6iw4bp7erwiptrgqd.onion",
+    "776aegl7tfhg6oiqqy76jnwrwbvcytsx2qegcgh2mjqujll4376ohlid.onion",
+    "77mdte42srl42shdh2mhtjr7nf7dmedqrw6bkcdekhdvmnld6ojyyiad.onion",
+    "azbpsh4arqlm6442wfimy7qr65bmha2zhgjg7wbaji6vvaug53hur2qd.onion",
+    "b64xcbleqmwgq2u46bh4hegnlrzzvxntyzbmucn3zt7cssm7y4ubv3id.onion",
+    "bsqbtcparrfihlwolt4xgjbf4cgqckvrvsfyvy6vhiqrnh4w6ghixoid.onion",
+    "bsqbtctulf2g4jtjsdfgl2ed7qs6zz5wqx27qnyiik7laockryvszqqd.onion",
+    "cwi3ekrwhig47dhhzfenr5hbvckj7fzaojygvazi2lucsenwbzwoyiqd.onion",
+    "devinbtcmwkuitvxl3tfi5of4zau46ymeannkjv6fpnylkgf3q5fa3id.onion",
+    "devinbtcyk643iruzfpaxw3on2jket7rbjmwygm42dmdyub3ietrbmid.onion",
+    "dtql5vci4iaml4anmueftqr7bfgzqlauzfy4rc2tfgulldd3ekyijjyd.onion",
+    "emzybtc25oddoa2prol2znpz2axnrg6k77xwgirmhv7igoiucddsxiad.onion",
```

Comment:
I think it would be better to have randomly created addresses (obviously with a valid checksum) instead of addrs that at sometime belonged to actual node runners on mainnet - less confusing (node runners may search for their node) and in case of a bug, likely no actual nodes are contacted.